catalog/imaging/imaging-plugin-nitf/src/test/groovy/org/codice/alliance/plugin/nitf/NitfPostProcessPluginSpec.groovy

```diff
@@ -0,0 +1,396 @@
+    }
+
+    def "handling processing large files in update requests"() {
+        setup:
+            nitfPostProcessPlugin.setMaxNitfSizeMB(4)
+        when: "a request contains a resource that is too large to process"
+            nitfPostProcessPlugin.processUpdate(updateProcessRequest)
+        then:
+            0 * metacard.setAttribute(_)
+    }
+
+    def "handling create requests with overview disabled"() {
+        setup:
+            nitfPostProcessPlugin.setCreateOverview(false)
+        when:
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            1 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value.toString().contains("original")})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+    }
+
+    def "handling update requests with overview disabled"() {
+        setup:
+            nitfPostProcessPlugin.setCreateOverview(false)
+        when:
+            nitfPostProcessPlugin.processUpdate(updateProcessRequest)
+        then:
+            1 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value.toString().contains("original")})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+    }
+
+    def "handling create requests with original disabled"() {
+        setup:
+            nitfPostProcessPlugin.setStoreOriginalImage(false)
+        when:
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            1 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value.toString().contains("overview")})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+    }
+
+    def "handling update requests with original disabled"() {
+        setup:
+            nitfPostProcessPlugin.setStoreOriginalImage(false)
+        when:
+            nitfPostProcessPlugin.processUpdate(updateProcessRequest)
+        then:
+            1 * metacard.setAttribute({it.name = Core.DERIVED_RESOURCE_URI; it.value.toString().contains("overview")})
+            1 * metacard.setAttribute({it.name == Core.THUMBNAIL; it.value != null})
+    }
+
+    def "handling non-nitf resources in create requests"() {
+        when:
+            nitfPostProcessPlugin.processCreate(createProcessRequest)
+        then:
+            processResource.getMimeType() >> "text/xml"
+            0 * metacard.setAttribute(_)
+    }
+
+    def "handling non-nitf resources in update requests"() {
+        when:
+            nitfPostProcessPlugin.processUpdate(updateProcessRequest)
+        then:
+            processResource.getMimeType() >> "text/xml"
+            0 * metacard.setAttribute(_)
+    }
+
+    def "handling metacards containing resource uri attribute during create requests"() {
+        setup:
+            def mockAttribute = Mock(AttributeImpl)
+        when: "a metacard has an existing resource uri attribute"
+            mockAttribute.name >> "fake"
```

Comment:
✏️ I know that this is just mocked here, but I think it's weird this name isn't `Core.DERIVED_RESOURCE_URI`. The javadoc for `getAttribute` states that the argument is the name of the `Attribute`. 
https://github.com/codice/ddf/blob/ddf-2.11.6/catalog/core/catalog-core-api/src/main/java/ddf/catalog/data/Metacard.java#L229-L235